bin/alluxio

```diff
@@ -46,77 +46,92 @@ function bootstrapConf {
     TOTAL_MEM=$(sysctl hw.memsize | cut -d ' ' -f2)
-    TOTAL_MEM=$[TOTAL_MEM / 1024]
+    TOTAL_MEM=$[TOTAL_MEM / 1024 / 1024]
     TOTAL_MEM=$[TOTAL_MEM * 2 / 3]
+    RAMDISK_FOLDER="/Volumes/ramdisk"
   else
-    ALLUXIO_SED="sed -i"
+    # Assuming Linux
     TOTAL_MEM=$(awk '/MemTotal/{print $2}' /proc/meminfo)
-    TOTAL_MEM=$[TOTAL_MEM * 2 / 3]
+    TOTAL_MEM=$[TOTAL_MEM / 1024 * 2 / 3]
+    RAMDISK_FOLDER="/mnt/ramdisk"
   fi
 
-  # Create a default config that can be overridden later
-  cp "${ALLUXIO_CONF_DIR}/alluxio-env.sh.template" ${dst}
-
   # Set the master hostname
-  ${ALLUXIO_SED} "s/XXX_MASTER_ADDRESS_XXX/$1/" ${dst}
+  ALLUXIO_MASTER_ADDRESS=${master}
 
   # Set the worker memory size
-  ${ALLUXIO_SED} "s/XXX_MEMORY_SIZE_XXX/${TOTAL_MEM}KB/" ${dst}
+  ALLUXIO_WORKER_MEMORY_SIZE="${TOTAL_MEM}MB"
+
+  # Set the ramdisk folder
+  ALLUXIO_RAM_FOLDER="${RAMDISK_FOLDER}"
 
   # Set under storage specific configuration
   case ${ufs} in
     local)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/\$\{ALLUXIO_HOME\}\/underFSStorage/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="\${ALLUXIO_HOME}/underFSStorage/"
       ;;
     hdfs)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/hdfs:\/\/localhost:9000\/alluxio/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="hdfs://localhost:9000/alluxio/"
       ;;
     s3)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/s3n:\/\/bucket\/directory/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="s3n://bucket/directory/"
       ;;
     gcs)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/gs:\/\/bucket\/directory/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="gs://bucket/directory/"
       ;;
     glusterfs)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/glusterfs:\/\/\//" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
-      export ALLUXIO_UNDERFS_GLUSTER_VOLUMES=\$\{ALLUXIO_UNDERFS_GLUSTER_VOLUMES:-alluxio_vol\}\\
-      " ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
-      export ALLUXIO_UNDERFS_GLUSTER_MOUNTS=\${ALLUXIO_UNDERFS_GLUSTER_MOUNTS:-\/vol\}\\
-      " ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
-      export ALLUXIO_UNDERFS_GLUSTER_MR_DIR=\${ALLUXIO_UNDERFS_GLUSTER_MR_DIR:-glusterfs:\/\/\/mapred\/system\}\\
-      " ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="/gluster_vol/"
       ;;
     swift)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/swift:\/\/swift-container/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="swift://swift-container/"
       ;;
     *)
       rm ${dst}
       echo ${usage}
       exit 1
       ;;
   esac
+
+  cat ${src} | while read -r line
+  do
+    case ${line} in
```

Comment:
- I will address the property file in later PR (with splitting the property files into multiple).
- One advantage of catting the template file is that the generated file is well documented, e.g., with instructions about the property files, with content like this (which I think will be helpful). Which is the risk in your mind to have the dependency on template?
```bash
#!/usr/bin/env bash

# Copy it as alluxio-env.sh and edit that to configure Alluxio for your
# site. This file is sourced to launch Alluxio servers or use Alluxio shell
# commands.
#
# This file provides one way to configure Alluxio options by setting the
# following listed environment variables. Note that, setting this file will not
# affect jobs (e.g., Spark job or MapReduce job) that are using Alluxio client
# as a library. Alternatively, you can edit alluxio-site.properties file, where
# you can set all the configuration options supported by Alluxio
# (http://alluxio.org/documentation/) which is respected by both external jobs
# and Alluxio servers (or shell).

# The directory where Alluxio deployment is installed
# ALLUXIO_HOME

# The directory where log files are stored. (Default: ${ALLUXIO_HOME}/logs).
# ALLUXIO_LOGS_DIR

# Address of the master.
ALLUXIO_MASTER_ADDRESS=localhost

# The directory where a worker stores in-memory data. (Default: /mnt/ramdisk).
# E.g. On linux, /mnt/ramdisk for ramdisk, /dev/shm for tmpFS; on MacOS, /Volumes/ramdisk for ramdisk
ALLUXIO_RAM_FOLDER=/Volumes/ramdisk

# Address of the under filesystem address. (Default: ${ALLUXIO_HOME}/underFSStorage)
# E.g. "/my/local/path" to use local fs, "hdfs://localhost:9000/alluxio" to use a local hdfs
ALLUXIO_UNDERFS_ADDRESS=${ALLUXIO_HOME}/underFSStorage/

# How much memory to use per worker
# E.g. "1000mb", "2gb"
ALLUXIO_WORKER_MEMORY_SIZE=10922MB

# Config properties set for Alluxio master daemon. (Default: "")
# E.g. "-Dalluxio.master.port=39999"
# ALLUXIO_MASTER_JAVA_OPTS

# Config properties set for Alluxio worker daemon. (Default: "")
# E.g. "-Dalluxio.worker.port=49999"
# ALLUXIO_WORKER_JAVA_OPTS

# Config properties set for Alluxio shell. (Default: "")
# E.g. "-Dalluxio.user.file.writetype.default=THROUGH"
# ALLUXIO_USER_JAVA_OPTS
```